### ISSUE TITLE
Support user name and email in gitconfig

### ISSUE THREAD

**Issue by metlos** [devfile/devworkspace-operator#694]
### Description
To achieve feature parity with the Che server, DWO should support setting the user name and email in the git config.

For Devworkspace, it seems reasonable to use the same approach as with other parts of the git configuration and have a dedicated label for marking a config map with this information. E.g.

```yaml
kind: ConfigMap
metadata:
  labels:
    "controller.devfile.io/git-user-config": "true"
    "controllder.devfile.io/watch-configmap": "true"
data:
  name: Will E. Coyote
  email: wcoyote@acme.com
```

UPDATE:
This issue has the following subtasks:
- [x] Allow injecting gitconfigs using a configmap ([PR](https://github.com/devfile/devworkspace-operator/pull/710)) (part of milestone 3)
- [ ] Retrieving user email and name tasks (not part of milestone 3):
  - [ ] https://github.com/eclipse/che/issues/21115 
  - [ ] https://github.com/eclipse/che/issues/20937
  - [ ] https://github.com/eclipse/che/issues/20938

**Comment by ibuziuk** [comment]
@l0rd @metlos the issue has been added to the endgame plan recently. is it expected to be there or could it sleep for the future releases?

**Comment by metlos** [comment]
I'll let @l0rd judge this. I added it to the list just so that it is not forgotten about before we declare feature parity.

**Comment by l0rd** [comment]
We have discussed this issue today. We have identified 2 distinct problems:

1. Retrieve the user name and email before a workspace is started (from the OIDC provider, git services APIs, asking the user etc...)
2. Use a `ConfigMap` to define user git configuration that works for both command line git and editors git UI

If 2 is implemented as in https://github.com/devfile/devworkspace-operator/pull/708 (mounted as `/etc/gitconfig` in all containers of the workspace pod) then the behavior is correct:
- If `/etc/gitconfig` exists, Theia loads values from it at statup time
- Setting Theia git username and email preferences adds them into `~/.gitconfig` in theia-ide container (that have priority over `/etc/gitconfig`)

For 1. in priority order, `che-server/che-operator` should try to get username and email:
- from the OIDC provider when creating the namespace
- from the git service provider (GitHub, Gitlab, Bitbucket) when creating a workspace
- from the git service provider (GitHub, Gitlab, Bitbucket) triggered by the user clicking on a new button in Dashboard("login with <service-provider>" button)
